api/src/main/java/org/pmiops/workbench/genomics/GenomicExtractionService.java

```diff
@@ -274,4 +276,26 @@ public GenomicExtractionJob submitGenomicExtractionJob(DbWorkspace workspace, Db
 
     return new GenomicExtractionJob().status(TerraJobStatus.RUNNING);
   }
+
+  public void abortGenomicExtractionJob(DbWorkspace dbWorkspace, String jobId) throws ApiException {
+    Optional<DbWgsExtractCromwellSubmission> dbSubmission =
+        wgsExtractCromwellSubmissionDao.findByWorkspaceWorkspaceIdAndWgsExtractCromwellSubmissionId(
+            dbWorkspace.getWorkspaceId(), Long.valueOf(jobId));
+
+    if (dbSubmission.isPresent()) {
```

Comment:
optional nit: personal preference, but I would probably invert this condition and throw, then eliminate the `else` block so the mainline codepath is not in a block.